Signal/src/Models/MessageActions.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+protocol MessageActionsDelegate: class {
+    func messageActionsShowDetailsForItem(_ conversationViewItem: ConversationViewItem)
+    func messageActionsReplyToItem(_ conversationViewItem: ConversationViewItem)
+}
+
+struct MessageActionBuilder {
+    static func reply(conversationViewItem: ConversationViewItem, delegate: MessageActionsDelegate) -> MenuAction {
+        return MenuAction(image: #imageLiteral(resourceName: "ic_reply"),
```

Comment:
RE: image literals: if you want to navigate from this usage of the image to the asset in xcassets, how do you do that?

Reply:
Although rendering the glyph into XCode is cute, I'm not really **that** into it. 

I primarily like it for:
- autocompletes image names
- it asserts that the file exists.